lib/ansible/module_utils/k8s/raw.py

```diff
@@ -60,24 +61,31 @@ def __init__(self, *args, **kwargs):
                 'apiVersion': api_version,
                 'metadata': {
                     'name': name,
-                    'namespace': namespace
+                    'namespace': namespace,
+                    'labels': labels
                 }
             }]
 
     def execute_module(self):
         changed = False
+        facts = False
         results = []
         self.client = self.get_api_client()
         for definition in self.resource_definitions:
             kind = definition.get('kind')
             search_kind = kind
             if kind.lower().endswith('list'):
                 search_kind = kind[:-4]
+                self.deprecate("Use the k8s_facts module with kind={0} rather than {1} with "
```

Comment:
How would this deprecation work if we implement kind=*List the same way other clients do (by processing each element in the list and doing a create/get/delete/update)?

Reply:
Do you think that's a thing that's likely to happen? Right now, the List mechanism is only documented to return facts, and causes other issues when used with patching (hence my original bug report in #41088). If there's a genuine plan to repurpose `List` for patching purposes, we can not deprecate, but I'd favour using k8s_facts to get the list, and then applying k8s with a loop.

Reply:
yeah I think *List will definitely get implemented, I haven't found the time yet but the goal is to match existing clients to maximize portability.